tests/test_messages.py

```diff
@@ -27,101 +33,129 @@ class WhiteCat(BaseModel):
 class DeepBody(BaseModel):
     deep_body: str = Field(..., min_length=20)
     url: AnyHttpUrl
+    ipv4: IPv4Address
+
+    @field_validator('url')
+    @classmethod
+    def validate_url(cls, v: AnyHttpUrl):
+        return v.__str__()
+
+    @field_validator('ipv4')
+    @classmethod
+    def validate_ipv4(cls, v: IPv4Address):
+        return v.__str__()
 
 
 class InnerNestedExample(BaseModel):
-    inner_body: List[DeepBody] = Field(..., max_items=2)
+    inner_body: List[DeepBody] = Field(..., max_length=2)
     cat: Cat
+    ipv6: IPv6Address
+
+    @field_validator('ipv6')
+    @classmethod
+    def validate_ipv6(cls, v: IPv6Address):
+        return v.__str__()
+
+
+CustomType = Annotated[str | int, Field(..., min_length=2, max_length=5)]
 
 
 class NestedExample(BaseModel):
     body: str = Field(..., max_length=2)
     inner: InnerNestedExample
+    custom_input: CustomType
+
+
+class MyEnum(str, Enum):
+    option = 'option'
+    choice = 'choice'
 
 
 class Example(BaseModel):
     string: str = Field(max_length=10)
-    integer: int = Field(default=1, min=10)
+    integer: int = Field(default=1)
     nested: NestedExample
+    ipt_enum: MyEnum
+
+    @field_validator('ipt_enum')
+    @classmethod
+    def validate_ipt_enum(cls, v: MyEnum):
+        return v.value
 
 
 def test_success():
     data = {
-        'string': 'string',
-        'integer': 1,
-        'nested': {
-            'body': 'st',
-            'inner': {
-                'inner_body': [{
-                    'deep_body': 'stringstringstringst',
-                    'url': 'https://example.com/'
+        "string": "string",
+        "integer": 1,
+        "nested": {
+            "body": "st",
+            "inner": {
+                "inner_body": [{
+                    "deep_body": "stringstringstringst",
+                    "url": "https://example.com/",
+                    "ipv4": "198.51.100.42"
                 }],
-                'cat': {
-                    'pet_type': 'cat',
-                    'color': 'black',
-                    'black_name': 'string'
-                }
-            }
-        }
+                "cat": {
+                    "pet_type": "cat",
+                    "color": "black",
+                    "black_name": "string"
+                },
+                "ipv6": "2001:db8:5b96::426f:8e17:642a"
+            },
+            "custom_input": "strin"
+        },
+        "ipt_enum": "option"
     }
-    assert Example(**data).dict() == data  # type: ignore
+    assert Example(**data).model_dump() == data  # type: ignore
 
 
 def test_i18n_validation():
```

Comment:
**Suggestion:** Use descriptive names for test functions to make it clear what each test is for.
```suggestion
def test_example_model_success():
    ...

def test_example_model_i18n_validation():
    ...
```